gensim/models/tfword2vec.py

```diff
@@ -0,0 +1,178 @@
+
+        min_count: The minimum number of word occurrences for it to be included in the vocabulary.
+
+        subsample: Subsample threshold for word occurrence. Words that appear with higher frequency
+                   will be randomly down-sampled. Set to 0 to disable.
+        """
+
+        if train_data is None:
+            raise ValueError("train_data must be specified.")
+
+        # Model options.
+
+        # Embedding dimension.
+        self.emb_dim = embedding_size
+
+        # Training options.
+
+        # The training text file.
+        self.train_data = train_data
+
+        # Number of negative samples per example.
+        self.num_samples = num_neg_samples
+
+        # The initial learning rate.
+        self.learning_rate = learning_rate
+
+        # Number of epochs to train. After these many epochs, the learning
+        # rate decays linearly to zero and the training stops.
+        self.epochs_to_train = epochs_to_train
+
+        # Concurrent training steps.
+        self.concurrent_steps = concurrent_steps
+
+        # Number of examples for one training step.
+        self.batch_size = batch_size
+
+        # The number of words to predict to the left and right of the target word.
+        self.window_size = window_size
+
+        # The minimum number of word occurrences for it to be included in the
+        # vocabulary.
+        self.min_count = min_count
+
+        # Subsampling threshold for word occurrence.
+        self.subsample = subsample
+
+        # Where to write out summaries.
+        self.save_path = save_path
+
+        # Eval options.
+
+        # The text file for eval.
+        self.eval_data = eval_data
+
+
+def modified_tfw2v_init(self, options, session):
+    self._options = options
+    self._session = session
+    self._word2id = {}
+    self._id2word = []
+    self.build_graph()
+    self.build_eval_graph()
+    if options.save_path is not None:
+        self.save_vocab()
+    if options.eval_data is not None:
+        self.read_analogies()
+
+
+tfw2v.__init__ = modified_tfw2v_init
+
+
+class TfWord2Vec(KeyedVectors):
+
+    def __init__(self, train_data=None, save_path=None, eval_data=None,
+        embedding_size=200, epochs_to_train=15, learning_rate=0.025,
+        num_neg_samples=25, batch_size=500, concurrent_steps=12,
+        window_size=5, min_count=5, subsample=1e-3, gpus=4):
+
+        self.options = Options(train_data, save_path=save_path, eval_data=eval_data,
+            embedding_size=embedding_size, epochs_to_train=epochs_to_train,
+            learning_rate=learning_rate, num_neg_samples=num_neg_samples,
+            batch_size=batch_size, concurrent_steps=concurrent_steps,
+            window_size=window_size, min_count=min_count, subsample=subsample)
+
+        self.convert_input(train_data)
+        self.gpus = gpus
+        self.train()
+        self.vocab = {}
+        self.create_vocab()
+
```

Comment:
implement a new saved method like in fasttext
https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/wrappers/fasttext.py#L57